src/app.js

```diff
@@ -269,43 +289,56 @@ var run = function () {
     var name = $(this).parent().find('.name').text()
 
     if (confirm('Are you sure you want to remove: ' + name + ' from local storage?')) {
-      storage.remove(name)
-      editor.removeSession(name)
-      editor.setNextFile(name)
-      updateFiles()
+      files.remove(name)
+      switchToNextFile()
+      editor.discard(name)
     }
     return false
   })
 
   editor.event.register('sessionSwitched', updateFiles)
 
   function switchToFile (file) {
-    editor.setCacheFile(file)
-    updateFiles()
+    currentFile = file
+
+    if (files.isReadOnly(file)) {
+      editor.openReadOnly(file, files.get(file))
+    } else {
+      editor.open(file, files.get(file))
+    }
   }
 
+  function switchToNextFile () {
+    switchToFile(Object.keys(files.list())[0])
```

Comment:
> Are we sure that the array is always not empty?

It should not be empty on startup, but it could become empty, that's true. Let me fix that.

> don't get why this will switch to the next file

It switched to the first file in the list, not actually the next file. Should rename the method I guess.